### PR TITLE
Upgrade to rustc 1.0.0-nightly (c89de2c56 2015-03-28) (built 2015-03-29)

### PR DESCRIPTION
**Pull request by Ogeon**
Removed a few deprecated things, such as a bunch of `as_slice`. I decided to let most of the `char_at` and friends stay for a little longer.

The tests are, once again, experiencing the [#23110](https://github.com/rust-lang/rust/issues/23110) ICE and I'm not sure where it originates this time.

### CODE REVIEW

**Comment by Manishearth** [comment]
Travis is [ICEing](https://travis-ci.org/servo/html5ever/builds/56209902) :/

/me tries locally

**Comment by Ogeon** [comment]
Yes, as I said. I have the same ICE in my own project where I'm using html5ever. I have no Idea where it comes from. There are no multi line `#![feature(...)]`.

**Comment by Manishearth** [comment]
Oh, right. Hm

File: macros/src/lib.rs [hunk 1]
```diff
@@ -11,12 +11,12 @@
 #![crate_type="dylib"]
 
 #![feature(plugin_registrar, quote)]
-#![feature(rustc_private, core, std_misc)]
+#![feature(rustc_private, convert)]
```
Comment (Ygg01) [inline macros/src/lib.rs]: Hm, is this example of multi-line feature?

Comment (Ogeon) [inline macros/src/lib.rs]: I tried to change them. No difference. Besides, they are in the macro crate, which is built before the html5ever crate. I can still change it if it's desired.

Comment (Ygg01) [inline macros/src/lib.rs]: @Ogeon, I don't think the issue is necessarily with `#![feature]`. I've been hitting a similar bug, without using `#![feature]` at all and on older version of Rust. I suspect it's some underlying macro expansion error.

Comment (Ogeon) [inline macros/src/lib.rs]: Sounds reasonable.

**Comment by Ogeon** [comment]
This is really strange... My project builds now, without any ICE, when using these fixes, but these tests are still failing. Nothing changed in my code.

**Comment by Ygg01** [comment]
@Ogeon : Did you run the `make check examples `. That's what causes ICE for me :cry:

**Comment by Ogeon** [comment]
I did run `make check` and it failed, but I could successfully build [symbiosis](https://github.com/Ogeon/symbiosis), where I'm using html5ever. That's what's odd.

**Comment by kmcallister** [comment]
I think all these changes are on master now. I worked around the ICE building examples by disabling debug info.

**Pull request closed without merging**